crates/locutus-node/src/node/event_listener.rs

```diff
@@ -167,6 +188,13 @@ fn create_log(logs: &[MessageLog], log: EventLog) -> (MessageLog, ListenerLogId)
 }
 
 fn fuse_events_msg<'a>(mut put_ops: impl Iterator<Item = &'a EventKind>) -> Option<EventKind> {
+    let mut put_ops = put_ops.filter(|op| {
```

Comment:
do the filtering in the iterator being passed to the function (which is already filtering itself)